### PR TITLE
Overhauled logging system

### PR DESCRIPTION
**Pull request by LMLiam**
- Now uses a main class
- Added more customisation for end-users
- Got rid of unnecessary browser checks
- Cleaned up code

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,58 +1,235 @@
...
+    const inDate = this.displayDate || this.displayTime ? new Date() : null;
+    const date = this.displayDate
+      ? ` ${this.getDateStringified(inDate as Date)}`
+      : "";
+    const time = this.displayTime
+      ? ` ${this.getTimeStringified(inDate as Date)}`
+      : "";
+
+    console.log(
+      `\x1b[1m${backgroundColor[0]}${indicator}${date}${time} (${level}) in ${fileName}\x1b[0m\n\n`,
```
Comment (shaunchander) [inline src/index.ts]: Remove indexing into [0].

File: src/index.ts [hunk 2]
```diff
@@ -1,58 +1,235 @@
...
 
-export const info = <T>(arg: T) => {
-	log(arg, LogType.INFO);
-};
+  /**
+   * Recommended usage by power users only. For everyone else, use built in #set<type>() methods in builder pattern.
+   * @param {LeucineConfiguration} config
+   * @returns {this}
+   */
+  public configure(config: LeucineConfiguration): this {
```
Comment (shaunchander) [inline src/index.ts]: does `:this` get inferred in the return type here?

File: src/index.ts (hunk 1 above)
Comment (LMLiam) [reply src/index.ts]: Great thanks - didn't spot that!

File: src/index.ts (hunk 2 above)
Comment (LMLiam) [reply src/index.ts]: It does get inferred yes, it can be removed

**Pull request merged**